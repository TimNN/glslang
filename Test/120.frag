#version 120

float lowp;
float mediump;
float highp;

float precision;

in vec4 i;
out vec4 o;

uniform sampler2D s2D;
centroid varying vec2 centTexCoord;

uniform mat4x2 m;

struct s {
    float f;
};

void main()
{
    mat2x3 m23 = mat2x3(m);

    int a;
    bool b;
    s sv = s(a);
    float[2] ia = float[2](3, i.y);
    float f1 = 1;
    float f = a;
    f = a;
    ivec3 iv3;
    vec3 v3 = iv3;
    f = f + a;
    f = a - f;
    f += a;
    f = a - f;
    v3 *= iv3;
    v3 = iv3 / 2.0;
    v3 = 3.0 * iv3;
    v3 = 2 * v3;
    v3 = v3 - 2;
    if (f <  a ||
        a <= f ||
        f >  a ||
        f >= a ||
        a == f ||
        f != a);
    f = b ? a : f;
    f = b ? f : a;
    f = b ? a : a;
    s news = sv;
    
    i.xy + i.xyz;      // ERROR
    m * i.xyz;         // ERROR
    m + i;             // ERROR
    int aoeu = 1.0;    // ERROR
    f = b;             // ERROR
    f = a + b;         // ERROR
    f = b * a;         // ERROR
    b = a;             // ERROR
    b = b + f;         // ERROR
    f |= b;            // ERROR

    gl_FragColor = texture2D(s2D, centTexCoord);

    float flat;
    float smooth;
    float noperspective;
    float uvec2;
    float uvec3;
    float uvec4;
    //packed;     // ERROR, reserved word

    {
        mat4 m;
        vec4 v;
        bool b;
        gl_FragColor += b ? v : m;  // ERROR, types don't match around ":"
    }

    gl_FragColor.xr;    // ERROR, swizzlers not from same field space
    gl_FragColor.xyxyx.xy; // ERROR, cannot make a vec5, even temporarily
    centTexCoord.z;     // ERROR, swizzler out of range
    (a,b) = true;       // ERROR, not an l-value
}

float imageBuffer;
float uimage2DRect;

int main() {}           // ERROR
void main(int a) {}     // ERROR

const int a;            // ERROR

int foo(in float a);
int foo(out float a)    // ERROR
{
    return 3.2;         // ERROR
    foo(a);             // ERROR
}

bool gen(vec3 v)
{
    if (abs(v[0]) < 1e-4 && abs(v[1]) < 1e-4)
        return true;
}

void v1()
{
}

void v2()
{
    return v1();  // ERROR, no expression allowed, even though void
}
