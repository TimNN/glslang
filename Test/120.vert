#version 120

in vec4 i;                // ERROR
out vec4 o;               // ERROR

attribute vec2 attv2;
attribute vec4 attv4;
uniform sampler2D s2D;
invariant varying vec2 centTexCoord;
invariant gl_Position;
centroid gl_Position;     // ERROR
centroid centroid foo;    // ERROR
invariant gl_Position, gl_PointSize;

void main()
{
    centTexCoord = attv2; 
    gl_Position = attv4;

    vec3[12] a;
    vec4[a.length()] b;
    gl_Position = b[b.length()-1];

    float f[];
    int a1 = f.length();  // ERROR
    float f[7];
    int aa = f.length();
    int a2 = f.length;    // ERROR
    int a3 = f.length(a); // ERROR
    int a4 = f.flizbit;   // ERROR
    int a4 = f.flizbit(); // ERROR
    float md[2][4];       // ERROR
    float[2] md2[4];      // ERROR
    float[2][4] md3;      // ERROR
    float md5, md6[2][3]; // ERROR
    float[2] md4, md7[4]; // ERROR
    float md9[2][3] = float[2][3](1, 2, 3, 4, 5, 6);  // ERROR
    float md10, md11[2][3] = float[2][3](1, 2, 3, 4, 5, 6);  // ERROR

    gl_PointSize = 3.8;
}

uniform float initted = 3.4;   // okay

const float concall = sin(0.3);

int[2][3] foo(                 // ERROR
              float[2][3] a,   // ERROR
              float[2] b[3],   // ERROR
              float c[2][3]);  // ERROR
