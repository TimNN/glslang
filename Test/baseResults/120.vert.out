ERROR: 0:3: 'in for stage inputs' : not supported for this version or the enabled extensions 
ERROR: 0:4: 'out for stage outputs' : not supported for this version or the enabled extensions 
ERROR: 0:11: 'gl_Position' : cannot add storage, auxiliary, memory, interpolation, or precision qualifier to an existing variable 
ERROR: 0:12: '' : can only have one auxiliary qualifier (centroid, patch, and sample) 
ERROR: 0:12: '' : replicated qualifiers 
ERROR: 0:12: 'foo' : identifier not previously declared 
ERROR: 0:25: 'length' :  array must be declared with a size before using this method
ERROR: 0:28: 'length' : incomplete method syntax 
ERROR: 0:29: 'length' : method does not accept any arguments 
ERROR: 0:30: 'flizbit' : only the length method is supported for array 
ERROR: 0:30: '=' :  cannot convert from '7-element array of float' to 'int'
ERROR: 0:31: 'flizbit' : only the length method is supported for array 
ERROR: 0:31: 'f' : no matching overloaded function found 
ERROR: 0:31: 'a4' : redefinition 
ERROR: 0:32: 'arrays of arrays' : not supported with this profile: none
ERROR: 0:33: 'arrays of arrays' : not supported with this profile: none
ERROR: 0:34: 'arrays of arrays' : not supported with this profile: none
ERROR: 0:35: 'arrays of arrays' : not supported with this profile: none
ERROR: 0:36: 'arrays of arrays' : not supported with this profile: none
ERROR: 0:37: 'arrays of arrays' : not supported with this profile: none
ERROR: 0:37: 'constructor' : array constructor needs one argument per array element 
ERROR: 0:37: 'arrays of arrays' : not supported with this profile: none
ERROR: 0:38: 'arrays of arrays' : not supported with this profile: none
ERROR: 0:38: 'constructor' : array constructor needs one argument per array element 
ERROR: 0:38: 'arrays of arrays' : not supported with this profile: none
ERROR: 0:47: 'arrays of arrays' : not supported with this profile: none
ERROR: 0:48: 'arrays of arrays' : not supported with this profile: none
ERROR: 0:49: 'arrays of arrays' : not supported with this profile: none
ERROR: 0:50: 'arrays of arrays' : not supported with this profile: none
ERROR: 29 compilation errors.  No code generated.

ERROR: node is still EOpNull!
0:15  Function Definition: main( (void)
0:15    Function Parameters: 
0:17    Sequence
0:17      move second child to first child (2-component vector of float)
0:17        'centTexCoord' (invariant smooth out 2-component vector of float)
0:17        'attv2' (in 2-component vector of float)
0:18      move second child to first child (4-component vector of float)
0:18        'gl_Position' (invariant gl_Position 4-component vector of float)
0:18        'attv4' (in 4-component vector of float)
0:22      move second child to first child (4-component vector of float)
0:22        'gl_Position' (invariant gl_Position 4-component vector of float)
0:22        direct index (4-component vector of float)
0:22          'b' (12-element array of 4-component vector of float)
0:22          11 (const int)
0:25      Sequence
0:25        move second child to first child (int)
0:25          'a1' (int)
0:25          1 (const int)
0:27      Sequence
0:27        move second child to first child (int)
0:27          'aa' (int)
0:27          7 (const int)
0:28      Sequence
0:28        move second child to first child (int)
0:28          'a2' (int)
0:29      Sequence
0:29        move second child to first child (int)
0:29          'a3' (int)
0:29          12 (const int)
0:37      Sequence
0:37        move second child to first child (2-element array of float)
0:37          'md9' (2-element array of float)
0:37          Construct float (const 2-element array of float)
0:38      Sequence
0:38        move second child to first child (2-element array of float)
0:38          'md11' (2-element array of float)
0:38          Construct float (const 2-element array of float)
0:40      move second child to first child (float)
0:40        'gl_PointSize' (invariant gl_PointSize float)
0:40        3.800000
0:43  Sequence
0:43    move second child to first child (float)
0:43      'initted' (uniform float)
0:43      3.400000
0:?   Linker Objects
0:?     'i' (in 4-component vector of float)
0:?     'o' (smooth out 4-component vector of float)
0:?     'attv2' (in 2-component vector of float)
0:?     'attv4' (in 4-component vector of float)
0:?     's2D' (uniform sampler2D)
0:?     'centTexCoord' (invariant smooth out 2-component vector of float)
0:?     'initted' (uniform float)
0:?     'concall' (const float)

